pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -106,11 +106,18 @@ func (a *Assignment) ToAPI() []kueue.PodSetAssignment {
 	return psFlavors
 }
 
+// TotalRequestsFor - returns the total quota needs of the wl, taking into account the potential
+// scaling needed in case of partial admission.
 func (a *Assignment) TotalRequestsFor(wl *workload.Info) resources.FlavorResourceQuantities {
 	usage := make(resources.FlavorResourceQuantities)
 	for i, ps := range wl.TotalRequests {
+		// in case of partial admission scale down the quantity
+		aps := a.PodSets[i]
+		if aps.Count != ps.Count {
```

Comment:
Please drop this check, as suggested in the [comment](https://github.com/kubernetes-sigs/kueue/pull/2826/files#r1719937771) it is not necessary as `ScaledTo` already takes care of it.

Reply:
`ScaledTo` need to do that after cloning the input , otherwise it will need to return a clone or the original depending on some opaque logic. 

At least for now `aps.Count != ps.Count` it's very unlikely to happen , the feature is supported by batch/Job only, additionally potential new frameworks supporting this will only allow one podSet to have a variable count, therefore by keeping this we avoid unnecessary allocations.

Reply:
I think it is a premature optimization. Clean code is more important in this case that the allocation.

Reply:
Unfortunately we need a clone every time we are scaling, otherwise we end up mutating the `Info.TotalRequests` that is used to re-queue the workload.

- We could ignore this aspect but we it could have surprising and very hard to debug side effects in the future.
- We could attempt to restore the scaling before re-queuing.

But any of this options are more complex or harder to follow the the current approach.

Reply:
Right.
I'm ok with keeping the optimization.